File: dashboard/src/api/unifiedAssets.ts
import { api } from "./client";

export interface UnifiedAsset {
  unified_id: string;
  noc_device_id: string | null;
  ipam_address_id: number | null;
  hostname: string | null;
  ip_address: string | null;
  mac_address: string | null;
  device_type: string | null;
  site: string | null;
  noc_status: string | null;
  ipam_status: string | null;
  region_name: string | null;
  cidr_block: string | null;
  os_guess: string | null;
  discovery_documented: boolean | null;
  discovery_open_ports: number;
  discovery_meta: Record<string, unknown> | null;
  criticality: string | null;
  registry_type: string | null;
  registry_sensor_key: string | null;
  ipam_linked: boolean;
  last_seen_at: string | null;
}

export interface UnifiedAssetPage {
  total: number;
  limit: number;
  offset: number;
  data: UnifiedAsset[];
}

export async function fetchUnifiedAssets(params?: {
  search?: string;
  linked_only?: boolean;
  limit?: number;
  offset?: number;
}) {
  const { data } = await api.get<UnifiedAssetPage>("/api/v1/ipam/assets/unified", { params });
  return data;
}

File: ipam/app/schemas/unified_assets.py
from pydantic import BaseModel, ConfigDict


class UnifiedAssetResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    unified_id: str
    noc_device_id: str | None = None
    ipam_address_id: int | None = None
    hostname: str | None = None
    ip_address: str | None = None
    mac_address: str | None = None
    device_type: str | None = None
    site: str | None = None
    noc_status: str | None = None
    ipam_status: str | None = None
    region_name: str | None = None
    cidr_block: str | None = None
    os_guess: str | None = None
    discovery_documented: bool | None = None
    discovery_open_ports: int = 0
    discovery_meta: dict | None = None
    criticality: str | None = None
    registry_type: str | None = None
    registry_sensor_key: str | None = None
    ipam_linked: bool = False
    last_seen_at: str | None = None


class UnifiedAssetPage(BaseModel):
    total: int
    limit: int
    offset: int
    data: list[UnifiedAssetResponse]
